src/main/scala/parsec/HListParsers.scala

```diff
@@ -1,63 +1,81 @@
 package parsec
 
 trait HListParsers extends Parsers with RepetitionParsers {
+
   import shapeless._
   import ops.hlist._
+  import syntax.singleton._
 
+  /* Define Boolean on the type level */
+  val (wTrue, wFalse) = (true.witness, false.witness)
+  type True = wTrue.T
+  type False = wFalse.T
 
-  /**
-   * Proof that parsing an HList of parsers
-   * yields a result of parsers
-   */
-  trait Parsable[L <: HList, Res <: HList] {
+  /** Proof that parsing an HList of parsers yields a result */
+  trait Parsable[L <: HList, B <: HList, Res <: HList] {
     def parse(ps: L, in: Input): ParseResult[Res]
   }
 
-  /**
-   * Parsing a single parser gives its result
-   */
-  implicit def singleParsable[T] = {
-    new Parsable[Parser[T] :: HNil, T :: HNil] {
-      def parse(ps: Parser[T] :: HNil, in: Input): ParseResult[T :: HNil]
-        = ps.head(in).map { x => x :: HNil }
+  /** Parsing a single parser with True gives a result (base case proof) */
+  implicit def baseCaseParsableTrue[T] = {
+    new Parsable[Parser[T] :: HNil, True :: HNil, T :: HNil] {
+      def parse(ps: Parser[T] :: HNil, in: Input): ParseResult[T :: HNil] =
+        ps.head(in).map { x => x :: HNil }
+    }
+  }
+
+  /** Parsing a single parser with False gives a result (base case proof) */
+  implicit def baseCaseParsableFalse[T] = {
+    new Parsable[Parser[T] :: HNil, False :: HNil, HNil] {
+      def parse(ps: Parser[T] :: HNil, in: Input): ParseResult[HNil] =
+        ps.head(in).map { _ => HNil }
     }
   }
 
-  /**
-   * Parsing recursively works too
-   */
-  implicit def recParsable[T, TS <: HList, Res <: HList]
-                          (implicit parsable: Parsable[TS, Res]) = {
-    new Parsable[Parser[T] :: TS, T :: Res] {
-      def parse(ps: Parser[T] :: TS, in: Input): ParseResult[T :: Res] = {
-        val (hd, tl) = (ps.head, ps.tail)
-        hd(in) match {
-          case f @ Failure(_, _) => f
-          case Success(a, rest) => parsable.parse(tl, rest) match {
-            case f @ Failure(msg, _) => Failure(msg, in)
-            case Success(as, rest2) => Success(a :: as, rest2)
+  /** Parsing multiple parsers with a True attached will ignore
```

Comment:
Now this is nitpicking, and I'm sorry for that, but could we have multiline comments start on the second line? :-P